certified-connectors/DocuSignDemo/script.csx

```diff
@@ -1614,6 +1624,89 @@ public class Script : ScriptBase
       response.Content = new StringContent(body.ToString(), Encoding.UTF8, "application/json");
     }
 
+    if ("GetRecipientFields".Equals(this.Context.OperationId, StringComparison.OrdinalIgnoreCase))
+    {
+      var body = ParseContentAsJObject(await response.Content.ReadAsStringAsync().ConfigureAwait(false), false);
+      var signers = body["signers"] as JArray;
+      var query = HttpUtility.ParseQueryString(this.Context.Request.RequestUri.Query);
+
+      var newSigner = new JObject();
+      var recipientEmailId = query.Get("recipientEmail");
+      var phoneNumber = query.Get("areaCode") + " " + query.Get("phoneNumber");
+      var signerPhoneNumber = "";
+      char[] charsToTrimPhoneNumber = { '*', ' ', '\'', '/', '-', '(', ')', '+'};
+
+      if ((recipientEmailId == null) && (query.Get("phoneNumber") == null))
+      {
+        throw new ConnectorException(HttpStatusCode.BadRequest, "ValidationFailure: Please fill either Recipient Email or Phone Number to retrieve Recipient information");
+      }
+
+      if (recipientEmailId != null)
+      {
+        foreach(var signer in signers)
+        {
+          if (recipientEmailId.ToString().Equals(signer["email"].ToString()))
+          {
+            newSigner["recipientId"] = signer["recipientId"];
+            newSigner["routingOrder"] = signer["routingOrder"];
+            newSigner["roleName"] = signer["roleName"];
```

Comment:
This code is repeating in 3 places. Please DRY out.

Reply:
Done